bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/dto/VenstarFanModeSerializer.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.venstarthermostat.internal.dto;
+
+import java.lang.reflect.Type;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link VenstarFanModeSerializer} parses fan mode values
+ * from the REST API JSON.
+ *
+ * @author Matthew Davies - Initial contribution
+ */
+
+public class VenstarFanModeSerializer implements JsonDeserializer<VenstarFanMode> {
+    @Override
+    public VenstarFanMode deserialize(JsonElement element, Type arg1, JsonDeserializationContext arg2)
+            throws JsonParseException {
+        int key = element.getAsInt();
+        return VenstarFanMode.fromInt(key);
```

Comment:
As `VenstarFanMode.fromInt` can throw an `IllegalArgumentException`, either you have to catch it here or to add `throws IllegalArgumentException` to the method and catch it at the higher level.
The best option could be to catch the exception in this method and to log something and return null. But then please check that the caller is handling a null value.
This comment is applicable to all the `fromInt` methods and so it has to be changed at different places.

Reply:
@lolodomo I'm a bit lost here. Honestly, this is at the limit of my programming abilities. Happy to learn though....

I understand there are 2 options.
1. Add try/catch to all methods calling xxxx.fromInt(), then make sure any calling functions can handle a null return value. I can't figure out how to do this because there is a GsonBuilder variable in the VenstarThermostatHandler class, which uses the VenstarFanModeSerializer class, and the VenstarFanModeSerializer class and its functions don't seem to get called anywhere else - so where am I supposed to be checking for capability to handle null? 
2. Throw the IllegalArgumentException from the function and catch it higher up the chain. In that case, is the code below sufficient or do I need to add something else?

Excuse my questions. It's not easy to follow and the original code before I started working on it (and which I copy/pasted a lot) already contained some of these problems.

`public class VenstarFanModeSerializer implements JsonDeserializer<VenstarFanMode> {
    @Override
    public VenstarFanMode deserialize(JsonElement element, Type arg1, JsonDeserializationContext arg2)
            throws JsonParseException,IllegalArgumentException {
        int key = element.getAsInt();
        return VenstarFanMode.fromInt(key);
    }
}`

Reply:
I believe the proper solution is to catch `IllegalArgumentException` and then throw a `JsonParseException`.
Then your call to gson.fromJson will certainlya utomatically throw a JsonSyntaxException in such a case.
So your code should look like this:
```
    @Override
    public VenstarFanMode deserialize(JsonElement element, Type arg1, JsonDeserializationContext arg2)
            throws JsonParseException {
        int key = element.getAsInt();
        try {
            return VenstarFanMode.fromInt(key);
        } catch (IllegalArgumentException e) {
            throw new JsonParseException(e);
        }
    }
```
Letr me know if this is still not clear.